python/ray/tune/ray_trial_executor.py

```diff
@@ -590,11 +597,12 @@ def _find_item(self, dictionary, item):
         out = [rid for rid, t in dictionary.items() if t is item]
         return out
 
-    def stop_trial(self,
-                   trial,
-                   error=False,
-                   error_msg=None,
-                   destroy_pg_if_cannot_replace=True):
+    def stop_trial(
+            self,
+            trial: Trial,
+            error: Optional[bool] = False,
+            error_msg: Optional[str] = None,
+            destroy_pg_if_cannot_replace: Optional[bool] = True) -> None:
```

Comment:
Shouldn't it be just `bool`? `Optional` essentially means "None is also accepted" (and not "this is an optional argument", [confusingly](https://docs.python.org/3/library/typing.html#typing.Optional)), and I don't think we have any special behavior for None here. Same with `error` argument.

Reply:
Got it, good to know. Let me update them :)